### ISSUE TITLE
[error] -bash: ptcli: command not found

### ISSUE THREAD

**Issue by mjunaidit** [ahmadawais/PTCL-CLI-Bash#2]
So here's what I did in the **Terminal** on my **macOS Sierra** that produced the following error.

> Junaids-MacBook-Pro:~ junaid$ sudo wget -qO ptcli https://git.io/vPtPS && sudo chmod +x ./ptcli && sudo install ./ptcli /usr/local/bin/ptcli
> Password:
> Junaids-MacBook-Pro:~ junaid$ ptcli
> -bash: ptcli: command not found
> Junaids-MacBook-Pro:~ junaid$ 

I'm new to this Terminal stuff so don't really know if it's a silly mistake on my side of some sort or if I'm using it all wrong.

**Comment by ahmadawais** [comment]
@mjunaidit Do you have `wget` installed? I am not sure if it comes packed with the new macOS.

**Comment by mjunaidit** [comment]
@ahmadawais Yeah I  installed `wget` from http://rudix.org/packages/wget.html

**Comment by ahmadawais** [comment]
@mjunaidit Can you run these commands one by one and paste the output here?

``` bash
sudo wget -qO ptcli https://git.io/vPtPS
```

``` bash
sudo chmod +x ./ptcli
```

``` bash
sudo install ./ptcli /usr/local/bin/ptcli
```

**Comment by mjunaidit** [comment]
> Junaids-MacBook-Pro:~ junaid$ sudo wget -qO ptcli https://git.io/vPtPS
> Password:
> Junaids-MacBook-Pro:~ junaid$ sudo chmod +x ./ptcli
> Junaids-MacBook-Pro:~ junaid$ sudo install ./ptcli /usr/local/bin/ptcli
> Junaids-MacBook-Pro:~ junaid$ 

If there was supposed to be some response, I'm getting nothing. Not an error nor a confirmation.

**Comment by MiteshShah** [comment]
@ahmadawais Your given short URL end with 404
https://git.io/vPtPS

I'd updated the link and I don't think we need install command to copy
 we can do that with wget command easily 

Pull request #3

**Comment by ahmadawais** [comment]
@MiteshShah  Oh boy! I had changed the name of that file and completely forgot about it. 
Install command is to help make this script executable by placing it in the user's $PATH since `/usr/local/bin/` is always in the path.

**Comment by ahmadawais** [comment]
@mjunaidit Try to install it now, it will work. Earlier the link was returning 404.

**Comment by mjunaidit** [comment]
<img width="697" alt="screen shot 2016-10-04 at 2 56 06 am" src="https://cloud.githubusercontent.com/assets/5680305/19055584/21b6a7c6-89de-11e6-9a12-0fef6fe24a1c.png">

I still get nothing no matter how I use the command. I do not get the error though now. But I don't see it working either.

> Junaids-MacBook-Pro:~ junaid$ sudo wget -qO ptcli https://git.io/vPmNx && sudo chmod +x ./ptcli && sudo install ./ptcli /usr/local/bin/ptcli
> Password:
> Junaids-MacBook-Pro:~ junaid$ ptcli
> Junaids-MacBook-Pro:~ junaid$ ptcli -c
> Junaids-MacBook-Pro:~ junaid$ ptcli

**Comment by ahmadawais** [comment]
@mjunaidit Use `ptcli` and not `ptcli -c`.
You should get prompts to enter data. Can you confirm if the CLI is installed by running the following commands and confirming that there is a `ptcli` file in there

``` bash
cd ~ && cd /usr/local/bin/ && ls -al
```

You should see a list of files, there should be `ptcli` in there. Let me know.

**Comment by MiteshShah** [comment]
> @MiteshShah Oh boy! I had changed the name of that file and completely forgot about it. 
> Install command is to help make this script executable by placing it in the user's $PATH since /usr/local/bin/ is always in the path.

@ahmadawais  @mjunaidit  We do not need install command as the following command do all the necessary steps

```
sudo wget -qO /usr/local/bin/ptcli https://git.io/vPmNx && sudo chmod +x /usr/local/bin/ptcli
```

![screen shot 2016-10-04 at 12 45 09 pm](https://cloud.githubusercontent.com/assets/1223371/19065312/789d890e-8a30-11e6-826e-92a7499c2a48.png)

**Comment by mjunaidit** [comment]
@ahmadawais Yep I do have that file in there!

<img width="697" alt="screen shot 2016-10-04 at 2 56 02 pm" src="https://cloud.githubusercontent.com/assets/5680305/19070008/c8e8b7dc-8a42-11e6-9194-02c704ed283b.png">

**Comment by MiteshShah** [comment]
@mjunaidit @ahmadawais  I'm sure inside that file no content.

Check what's inside of that file using `cat /usr/local/bin/ptcli`

**Comment by mjunaidit** [comment]
@MiteshShah if this means the file is empty, then it is!

> Junaids-MacBook-Pro:~ junaid$ cat /usr/local/bin/ptcli
> Junaids-MacBook-Pro:~ junaid$

**Comment by MiteshShah** [comment]
@mjunaidit run following command

```
rm -f /usr/local/bin/ptcli 
sudo wget -qO /usr/local/bin/ptcli https://git.io/vPmNx && sudo chmod +x /usr/local/bin/ptcli
```

This will fix your issue

**Comment by ahmadawais** [comment]
@MiteshShah I studied the command `install` it just copies it there with correct permissions if applied. You sure, there won't be an issue with directly downloading the file there? I think I am up for that edit now.

@mjunaidit Mitesh is right. Your file has nothing in it since our link was broken. 
Can you run the following command and try again?

``` bash
rm -f /usr/local/bin/ptcli 
sudo wget -qO /usr/local/bin/ptcli https://git.io/vPmNx && sudo chmod +x /usr/local/bin/ptcli
```

**Comment by MiteshShah** [comment]
@ahmadawais If you notice EasyEngine we mostly use wget to directly download on destination folder.
 Which saves few extra steps to copy files in that destination directory.

We does not need install command here.
Without it its work just fine refer above my attached screenshot.

**Comment by mjunaidit** [comment]
Now I get a permission denied error!

> Junaids-MacBook-Pro:~ junaid$ rm -f /usr/local/bin/ptcli 
> Junaids-MacBook-Pro:~ junaid$ sudo wget -qO /usr/local/bin/ptcli https://git.io/vPmNx && sudo chmod +x /usr/local/bin/ptcli
> Password:
> Junaids-MacBook-Pro:~ junaid$ ptcli
> -bash: /usr/local/bin/ptcli: Permission denied
> Junaids-MacBook-Pro:~ junaid$

**Comment by ahmadawais** [comment]
@mjunaidit 
OK, I think install method is better.

Just run `rm -f /usr/local/bin/ptcli` and then run this again 

``` bash
sudo wget -qO ptcli https://git.io/vPmNx && sudo chmod +x ./ptcli && sudo install ./ptcli /usr/local/bin/ptcli
```

**Comment by ahmadawais** [comment]
@MiteshShah Permissions are not a mess when it comes to Ubuntu where EE is used, this however is used in macOS and as from the last response of @mjunaidit  you can see what I was afraid would happen.

**Comment by jabranr** [comment]
For OS X, try using `cURL` instead of `wget` if that helps:

``` shell
curl https://git.io/vPmNx -L -o ptcli && chmod +x ptcli && sudo mv ptcli /usr/local/bin/ptcli
```

**Comment by ahmadawais** [comment]
@jabranr bhai the URL was returning 404 which downloaded a copy of blank page on @mjunaidit 's end.

**Comment by jabranr** [comment]
@ahmadawais oh I see. Didn't realise that was the case.
